src/main/java/org/broadinstitute/hellbender/utils/Utils.java

```diff
@@ -1274,4 +1275,70 @@ private static void removeTrailingEmptyStringsFromEnd(final List<String> result)
             result.remove(result.size() - 1);
         }
     }
+
+    /**
+     * Convenience function that formats the novelty rate as a %.2f string
+     *
+     * @param x number of objects part of total that meet some criteria
+     * @param total count of all objects, including x
+     * @return a String percent rate, or NA if total == 0
+     */
+    public static String formattedPercent(final long x, final long total) {
+        return total == 0 ? "NA" : String.format("%.2f", (100.0*x) / total);
+    }
+
+    /**
+     * Convenience function that formats a ratio as a %.2f string
+     *
+     * @param num  number of observations in the numerator
+     * @param denom number of observations in the denumerator
+     * @return a String formatted ratio, or NA if all == 0
+     */
+    public static String formattedRatio(final long num, final long denom) {
+        return denom == 0 ? "NA" : String.format("%.2f", num / (1.0 * denom));
+    }
+
+    /**
+     * Given a collection of strings and a collection of regular expressions, generates the set of strings that match
+     * any expression
+     * @param sourceValues collection of strings from which to to select
+     * @param filterExpressions list of expressions to use for matching
+     * @param exactMatch If true match filters exactly, otherwise use as both exact and regular expressions
+     * @return A new set strings from sourceValues that satisfy at least one of the expressions in sampleExpressions
+     */
+    public static Set<String> filterCollectionByExpressions(final Collection<String> sourceValues, final Collection<String> filterExpressions, final boolean exactMatch) {
+        if (filterExpressions == null) {
+            return Collections.emptySet();
+        }
+
+        Utils.nonNull(sourceValues);
+
+        final Set<String> filteredValues = new LinkedHashSet<>();
+
+        Collection<Pattern> patterns = null;
+        if (!exactMatch) {
+            patterns = compilePatterns(filterExpressions);
+        }
+        for (final String value : sourceValues) {
+            if (filterExpressions.contains(value)) {
+                filteredValues.add(value);
+            } else if (!exactMatch) {
+                for (final Pattern pattern : patterns) {
```

Comment:
As @cmnbroad mentioned, this loop should end after the first match.

Reply:
covered above